lib/index.js

```diff
@@ -364,33 +397,50 @@ function initDB(peer, config) {
 
   /**
    * @param {Array<MsgID>} tangleIDs
+   * @param {CB<Record<MsgID, DBTangle>>} cb
    */
-  function populateTangles(tangleIDs) {
+  function populateTangles(tangleIDs, cb) {
     /** @type {Record<MsgID, DBTangle>} */
     const tangles = {}
-    for (const tangleID of tangleIDs) {
-      tangles[tangleID] ??= new DBTangle(tangleID, records(), get)
-    }
-    return tangles
+    pull(
+      pull.values(tangleIDs),
+      pull.asyncMap((tangleID, cb) => {
+        DBTangle.init(tangleID, records(), get).then((dbtangle) => {
+          tangles[tangleID] ??= dbtangle
+          cb(null, null)
+        })
+      }),
+      pull.drain(
+        () => {},
+        () => {
+          return cb(null, tangles)
+        }
+      )
+    )
   }
 
   /**
    * @param {Pick<RecPresent, 'id' | 'msg'>} rec
-   * @returns {DBTangle | null}
+   * @param {(err: Error | null, tangle: DBTangle | null) => void} cb
```

Comment:
If i change this to
```
   * @param {CB<DBTangle | null>} cb
```

then i get this error further down

```
        return cb(null, accountTangle)
...
Argument of type '[null, DBTangle]' is not assignable to parameter of type '[Error] | [null, null]'.
  Type '[null, DBTangle]' is not assignable to type '[null, null]'.
    Type at position 1 in source is not compatible with type at position 1 in target.
      Type 'DBTangle' is not assignable to type 'null'.ts(2345)
(parameter) cb: (...args: [Error] | [null, null]) => void
@param cb
```

Reply:
This CB type you made is fine IMO